backend/test_frontend_streaming.py

```
@@ -0,0 +1,209 @@
+        }
+        
+        sse_line = f"data: {json.dumps(final_response)}\n\n"
+        print(f"[{time.time() - start_time:.2f}s] 📤 Frontend received: {sse_line.strip()}")
+        print(f"    🎯 Frontend: Displaying final response")
+        
+        return progress_updates, result
+    
+    # Run the simulation
+    try:
+        progress_updates, result = await simulate_frontend_processing()
+        total_time = time.time() - start_time
```

Ruff (0.12.2)

Undefined name `start_time`

(F821)